effekt/shared/src/main/scala/effekt/machine/Transformer.scala

```diff
@@ -139,7 +139,7 @@ object Transformer {
       case lifted.App(lifted.BlockVar(id), List(), args) =>
         // TODO deal with BlockLit
         id match {
-          case symbols.UserFunction(_, _, _, _, _, _, _) =>
+          case symbols.UserFunction(_, _, _, _, _, _, _)  | symbols.TmpBlock(_) =>
```

Comment:
The matching here on symbols here is a smell. What is it that you really need here?